src/serac/coefficients/coefficient_extensions.hpp

```diff
@@ -65,8 +197,49 @@ mfem::Array<int> MakeEssList(mfem::ParFiniteElementSpace& pfes, mfem::VectorCoef
  * attribute 2, otherwise attribute 1.
  * @return An array holding the attributes that correspond to each element
  */
-mfem::Array<int> MakeAttributeList(
-    mfem::Mesh& m, mfem::Coefficient& c, std::function<int(double)> digitize = [](double v) { return v > 0. ? 2 : 1; });
+
+template <typename T>
+T MakeAttributeList(mfem::Mesh& m, mfem::Coefficient& c, std::function<int(double)> digitize = digitize::threshold)
+{
+
+  static_assert( std::is_constructible<T, typename detail::index_t<T>::type>::value, "T does not have a constructor of type detail::index_t<T>");
```

Comment:
Suggestion: "T must be constructible with an instance of detail::index_t<T>"

Also, the `@pre` and `@tparam` Doxygen commands might be useful here for documenting template preconditions.